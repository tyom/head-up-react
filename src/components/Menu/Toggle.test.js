import React from 'react';
import Toggle from './Toggle';

const onClickSpy = jest.fn();

describe('Toggle: structure', () => {
  test('renders as open', () => {
    const component = shallow(<Toggle onClick={onClickSpy} />);
    expect(component).toMatchSnapshot();
  });

  test('renders as closed', () => {
    const component = shallow(
      <Toggle isMenuClosed={true} onClick={onClickSpy} />
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Toggle: behaviour', () => {
  test('responds to clicks', () => {
    const component = shallow(<Toggle onClick={onClickSpy} />);

    component.simulate('click');

    expect(onClickSpy).toBeCalled();
  });
});
