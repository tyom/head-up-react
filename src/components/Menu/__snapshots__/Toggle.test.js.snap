// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle: structure renders as closed 1`] = `
<button
  className="style_toggle__2daLT"
  onClick={[MockFunction]}
  type="button"
>
  <FaEllipsisV
    className="style_toggleIcon__3MeTu"
  />
  <span>
    Open
     menu
  </span>
</button>
`;

exports[`Toggle: structure renders as open 1`] = `
<button
  className="style_toggle__2daLT"
  onClick={[MockFunction]}
  type="button"
>
  <FaAngleLeft
    className="style_toggleIcon__3MeTu"
  />
  <span>
    Close
     menu
  </span>
</button>
`;
